app/dashboard/tests/test_dashboard_utils.py

```diff
@@ -73,3 +78,35 @@ def test_humanize_event_name():
         """Test the humanized representation of an event name."""
         assert humanize_event_name('start_work') == 'WORK STARTED'
         assert humanize_event_name('remarket_funded_issue') == 'REMARKET_FUNDED_ISSUE'
+
+    @staticmethod
+    @patch('dashboard.utils.UserAction.objects')
+    def test_create_user_action_with_cookie(mockUserAction):
+        """Test the giving utm* in cookie should store in DB."""
+        request = RequestFactory().get('/login')
+        request.COOKIES['utm_source'] = 'test source'
+        request.COOKIES['utm_medium'] = 'test medium'
+        request.COOKIES['utm_campaign'] = 'test campaign'
+        create_user_action(None, 'Login', request)
+        mockUserAction.create.assert_called_once_with(action='Login', metadata={}, user=None,
+                                                      utm={'utm_source': 'test source',
+                                                           'utm_medium': 'test medium',
+                                                           'utm_campaign': 'test campaign'})
+
+    @staticmethod
+    @patch('dashboard.utils.UserAction.objects')
+    def test_create_user_action_without_cookie(mockUserAction):
+        """Test the giving utm* in cookie should store in DB."""
+        request = RequestFactory().get('/login')
+        create_user_action(None, 'Login', request)
+        mockUserAction.create.assert_called_once_with(action='Login', metadata={}, user=None)
+
+    @staticmethod
+    @patch('dashboard.utils.UserAction.objects')
+    def test_create_user_action_campaign_not_json(mockUserAction):
```

Comment:
How is this test different then `test_create_user_action_with_cookie`?

Reply:
I want to test partial value in cookie scenario...

Reply:
Gotcha.

Does it make sense to rename the test method to something like `test_create_user_action_with_partial_cookie`?